### PR TITLE
[PN-11230] Feat: Added radd store registry Lambda function code

### PR DESCRIPTION
**Pull request by flaminiaScarciofolo**
## Short description

Include a summary of the changes.

## List of changes proposed in this pull request

- Feature A
- Feature B

## How to test

Describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

### CODE REVIEW

File: functions/raddStoreRegistryLambda/sonarqube-scanner.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const options = {
+  "sonar.organization": "pagopa",
+  "sonar.projectKey": "pagopa_pn-",
```
Comment (masciale89) [inline functions/raddStoreRegistryLambda/sonarqube-scanner.js:3]: @flaminiaScarciofolo creato progetto su sonar con la chiave `pagopa_pn-bff-raddStoreRegistryLambda` nome da utilizzare qui

File: functions/raddStoreRegistryLambda/src/app/ssmParameter.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      )}`,
+      {
+        headers: {
+          "X-Aws-Parameters-Secrets-Token": process.env.AWS_SESSION_TOKEN,
+        },
+      }
+    );
+    return response.data.Parameter.Value;
+  } catch (err) {
+    console.error("Unable to get secret ", err);
```
Comment (masciale89) [inline functions/raddStoreRegistryLambda/src/app/ssmParameter.js:49]: da cambiare secret con parameter

File: functions/raddStoreRegistryLambda/src/app/ssmParameter.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+const getParameter = async (parameterName) => {
+    console.log(`Retrieving parameter: ${parameterName}`);
+    const command = new GetParameterCommand({ Name: parameterName });
+    const response = await client.send(command);
+    console.log(`Successfully retrieved parameter: ${parameterName}`);
+    return response.Parameter.Value;
+}
+
+async function getParameterFromLayer(parameterName) {
```
Comment (AndreaCimini90) [inline functions/raddStoreRegistryLambda/src/app/ssmParameter.js:35]: Why do we use axios and not sdk?

File: functions/raddStoreRegistryLambda/src/app/s3Utils.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+const { S3Client, ListObjectVersionsCommand, CopyObjectCommand, PutObjectCommand } = require('@aws-sdk/client-s3');
```
Comment (AndreaCimini90) [inline functions/raddStoreRegistryLambda/src/app/s3Utils.js:1]: CopyObjectCommand and PutObjectCommand are unused

File: functions/raddStoreRegistryLambda/src/test/eventHandler.test.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+const ssmUtils = require('../app/ssmParameter');
+const utils = require('../app/utils');
+const sinon = require('sinon');
+const assert = require('node:assert/strict');
+let originalEnv;
+
+function setupEnv() {
+  // Mock the environment variables
+  process.env = {
+    ...originalEnv,
```
Comment (AndreaCimini90) [inline functions/raddStoreRegistryLambda/src/test/eventHandler.test.js:14]: Do we need this variable?

File: functions/raddStoreRegistryLambda/src/test/resources/mock.yaml [hunk 6]
```diff
@@ -0,0 +1,156 @@
+openapi: 3.0.1
```
Comment (AndreaCimini90) [inline functions/raddStoreRegistryLambda/src/test/resources/mock.yaml:1]: Why do we need this file?

File: functions/raddStoreRegistryLambda/src/test/ssmParameter.test.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+const { retrieveCsvConfiguration, retrieveGenerationConfigParameter } = require('../app/ssmParameter');
+const { expect } = require('chai');
+const axios = require('axios');
+var MockAdapter = require("axios-mock-adapter");
+const assert = require('node:assert/strict');
+
+
+describe('ssmParameter', () => {
+    let ssmClientMock;
+	beforeEach(() => {
```
Comment (AndreaCimini90) [inline functions/raddStoreRegistryLambda/src/test/ssmParameter.test.js:12]: fix the indentation

File: functions/raddStoreRegistryLambda/src/test/utils.test.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+const { checkIfIntervalPassed } = require('../app/utils');
+const { expect } = require('chai');
+
+describe('checkIfIntervalPassed', () => {
+  beforeEach(() => {
+    process.env.GENERATE_INTERVAL = '1';
+  });
+
+  it('returns true when interval has passed', () => {
+    let date = new Date();
```
Comment (AndreaCimini90) [inline functions/raddStoreRegistryLambda/src/test/utils.test.js:10]: use const

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**